### PR TITLE
Avoid errors unpacking ZIP file on PHP 7.0

### LINKED ISSUE #7313: php 7.0 error uploading & installing .zip packages
#### Description

2.1 doesn't like installing .zips on php 7.0, in either windows or linux.   

Note that php 7.3 & 7.4 work fine. Note that tar.gz works fine.

image

Error log:
image

### Steps to reproduce
1. Use php 7.0
2. Install a .zip package

### Environment (complete as necessary)
- Version/Git revision: current
- Database Type: mysql
- Database Version: 5.7
- PHP Version: 7.0

### Additional information/references
7.0 is the new 5.4...

### PR DESCRIPTION
**Pull request by Sesquipedalian**
(Maybe) fixes #7313

### CODE REVIEW

File: Sources/Subs-Package.php [hunk 1]
```diff
@@ -239,7 +239,7 @@ function read_zip_data($data, $destination, $single_file = false, $overwrite = f
 	$return = array();
 
 	// End of central directory record (EOCD)
-	$cdir = unpack('vdisk/@4/vdisk_entries/ventries/@12/Voffset', $data, $data_ecr + 4);
+	$cdir = unpack('vdisk/@4/vdisk_entries/ventries/@12/Voffset', substr($data, $data_ecr + 4));
```
Comment (live627) [inline Sources/Subs-Package.php]:
```suggestion
	$cdir = unpack('vdisk/@4/vdisk_entries/ventries/@12/Voffset', substr($data, $data_ecr + 4, 16));
```

File: Sources/Subs-Package.php [hunk 2]
```diff
@@ -251,7 +251,7 @@ function read_zip_data($data, $destination, $single_file = false, $overwrite = f
 	for ($i = 0; $i < $cdir['entries']; $i++)
 	{
 		// Central directory file header
-		$header = unpack('Vcompressed_size/@8/vlen1/vlen2/vlen3/vdisk/@22/Voffset', $data, $pos_entry + 20);
+		$header = unpack('Vcompressed_size/@8/vlen1/vlen2/vlen3/vdisk/@22/Voffset', substr($data, $pos_entry + 20));
```
Comment (live627) [inline Sources/Subs-Package.php]:
```suggestion
		$header = unpack('Vcompressed_size/@8/vlen1/vlen2/vlen3/vdisk/@22/Voffset', substr($data, $pos_entry + 20, 26));
```

File: Sources/Subs-Package.php [hunk 3]
```diff
@@ -263,8 +263,7 @@ function read_zip_data($data, $destination, $single_file = false, $overwrite = f
 		// Local file header (so called because it is in the same file as the data in multi-part archives)
 		$file_info = unpack(
 			'vflag/vcompression/vmtime/vmdate/Vcrc/Vcompressed_size/Vsize/vfilename_len/vextra_len',
-			$data,
-			$header['offset'] + 6
+			substr($data, $header['offset'] + 6)
```
Comment (live627) [inline Sources/Subs-Package.php]:
```suggestion
			substr($data, $header['offset'] + 6, 24)
```

File: Sources/Subs-Package.php [hunk 4]
```diff
@@ -285,7 +284,7 @@ function read_zip_data($data, $destination, $single_file = false, $overwrite = f
 			if (substr($data, $gplen, 4) === "\x50\x4b\x07\x08")
 				$gplen += 4;
 
-			if (($general_purpose = unpack('Vcrc/Vcompressed_size/Vsize', $data, $gplen)) !== false)
+			if (($general_purpose = unpack('Vcrc/Vcompressed_size/Vsize', substr($data, $gplen))) !== false)
```
Comment (live627) [inline Sources/Subs-Package.php]:
```suggestion
			if (($general_purpose = unpack('Vcrc/Vcompressed_size/Vsize', substr($data, $gplen, 12))) !== false)
```

**Review by live627 — CHANGES_REQUESTED** [review]
ad dbyte lengths to substrrings

**Comment by BrickOzp** [comment]
Works for me, php 7.0

**Comment by Sesquipedalian** [comment]
@live627: Done.

@BrickOzp: Does everything still work for you on PHP 7.0 with the changes in 98ed90c?

**Review by live627 — APPROVED** [review]

**Comment by BrickOzp** [comment]
Yes, it still works

**Pull request merged**